src/utils/ImportPathResolver.ts

```diff
@@ -32,15 +32,15 @@ import { resolve, join, dirname } from 'path';
 export type StringReplacer = (importStatement: string) => string;
 
 const anyQuote = '["\']';
-const notQuote = '[^"\']';
-const importString = `(?:${anyQuote}${notQuote}+${anyQuote})`;
+const pathStringContent = `[^"'\r\n]+`;
+const importString = `(?:${anyQuote}${pathStringContent}${anyQuote})`;
 
 // Separate patterns for each style of import statement,
 // wrapped in non-capturing groups,
 // so that they can be strung together in one big pattern.
 const funcStyle = `(?:\\b(?:import|require)\\s*\\(\\s*${importString}\\s*\\))`;
 const globalStyle = `(?:\\bimport\\s+${importString})`;
-const fromStyle = `(?:\\bfrom\\s+${notQuote}*?${importString})`;
+const fromStyle = `(?:\\bfrom\\s+${importString})`;
```

Comment:
This was the main culprit, mean to allow parens around import paths.

The fact that parens were allowed made it more difficult to crate a sufficiently restrictive path-matching regex. But having parens around import-paths strings (unless using functional importing, e.g. import("path")) is not valid according to spec. I think you can technically do it with CommonJS since imports are synchronous, but it's not allowed with Modules since imports are async.

Long story short: if we don't allow parens around import paths we don't lose any important functionality and restrictive matching is easier.